### DISCUSSION TITLE
Is it possible to train one class datasets?

### DISCUSSION THREAD

**Discussion by cs1488** [open-mmlab/mmdetection3d#2281] [Q&A] [answered]
Hello, 
i am currently trying to train DGCNN on a custom data set.
Since the data set used in the examples is S3DIS,  i chose to bring my custom data set into this format.
I want to do the semantic segmentation.

I only have one class (a simple cube)  in my dataset
However since the data converter for S3DIS just concatenates all the files stored in an annotation path i also
created an "clutter" class, to have points that do not belong to the cube.

I am asking myself, if it is even possible to train on a class that only got 1 class?

The conversion with the collect_indoor3d_data.py seemed to work fine,
also the conversion from middle format to final does not differ from what i have seen on S3DIS.

Sadly i can not browse the data sets, neither S3DIS nor mine with tools/misc/browse_dataset.py since both are missing PALETTE Keyword.

So maybe already the conversion failed.

I tried my best in writing the necessary config files with those of S3DIS as template.

But when training, the first thing the modell seems to try is:
just classify all points as the first class, being cube and then be done with it.

This way it achieves around 0.00644 in accuracy for the cube and 0 for clutter.

The value does not change over time.

I sorted my files in directories like they are used in S3DIS, instead of 6 Areas  my data set has 24 Folders.
On my first try i used 23 folders vor training and 1 for validation.

In the first epoch i got -0,004 in loss, which seems wrong to me.

Because training takes very long on these amount of data, and i wanted to change the parameters in the configs to find one that does anything i used a smaller set of images.
For these tries i can not report any loss.

NOTE: i needed to adjust the cross entropy function  in the openmmlab/Lib/site-packages/torch/nn/ functional.py
i got the 
``return torch._C._nn.cross_entropy_loss(input, target, weight, _Reduction.get_enum(reduction), ignore_index, label_smoothing)
RuntimeError: expected scalar type Long but found Int``
Error so i added
 ``target = target.to(dtype=torch.long)``

before the return line (arround line 3030)

Which was suggested here:
[#446 ](https://github.com/open-mmlab/mmtracking/issues/446)
I link my logs and reports below.
Negative Loss: 
[20230214_145708.log](https://github.com/open-mmlab/mmdetection3d/files/10744182/20230214_145708.log)
Smaller Dataset
[20230215_140648.log](https://github.com/open-mmlab/mmdetection3d/files/10744190/20230215_140648.log)

**Comment by JingweiZhang12** [comment, marked as answer]
Here is an example of only training `car` in kitti dataset: https://github.com/open-mmlab/mmdetection3d/blob/1.1/configs/_base_/datasets/kitti-3d-car.py#L7
